### DISCUSSION TITLE
Command window focus loss and refocus changes text selection

### DISCUSSION THREAD

**Discussion by codaroma** [arnoudkooi/SN-Utils#527] [Q&A]
Has there been a change in behaviour in a recent update? I have noticed that if I open the slash command popup, type the start of a command, then go to another tab or application to copy something, when I come back, the whole text in the command popup is selected so I have to change the selection before I can paste in the rest of my command. This causes me a lot of problems with my clipboard history application (CopyQ) which triggers a focus loss and then a regain of focus followed by a paste, which then overwrites everything in the command popup that I have already entered. Is this behaviour new? Is it optional?
A use case example:
I open the slash command popup via the hotkey, I type "u" command (or similar) followed by a space, I then call up my clipboard history app using its hotkey, I select an item from my clipboard history app which moves it into the OS clipboard, sends focus back to where I was before, and sends a paste command. Unfortunately the text selection has now changed and the paste overwrites the command I have already entered. This happens even if I am not using my clipboard history application. I might be going to notepad to get/copy the extra info I need instead.

**Comment by arnoudkooi** [comment]
I'll take a look. I think if you hit the right arrow before pasting, it also solves the issue.

**Reply by codaroma** [reply]
Yes, thanks for the suggestion. Unfortunately that is not an option with CopyQ as it sends the focus back, and sends the paste command all in one action. Plus that assumes I want to paste it at the end of what I have already typed. If I have already placed the cursor exactly where I want to insert my clipboard text before I go off to another app to go grab that text, then the right arrow trick is of no use if that cursor position is not at the end of the line.

**Reply by arnoudkooi** [reply]
Couldn't directly figure out why this happens, ill do some more troubleshooting

**Reply by vedang444** [reply]
@codaroma If you are using a Mac, you can try using Maccy instead. It doesn't paste the selection in one go.
For Windows, I just used the default OOB clipboard manager that you can enable in settings and use via Win+V. Let me know if that helps.
